presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkMaterializedRow.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.classloader_interface;
+
+import java.io.Serializable;
+import java.nio.ByteBuffer;
+
+import static java.util.Objects.requireNonNull;
+
+public class PrestoSparkMaterializedRow
+        implements Serializable
+{
+    private final byte[] data;
+
+    public PrestoSparkMaterializedRow(byte[] data)
+    {
+        this.data = requireNonNull(data, "data is null");
+    }
+
+    public byte[] getData()
+    {
+        return data;
+    }
+
+    public PrestoSparkMutableRow toPrestoSparkMutableRow()
+    {
+        PrestoSparkMutableRow prestoSparkMutableRow = new PrestoSparkMutableRow();
+        prestoSparkMutableRow.setBuffer(ByteBuffer.wrap(data));
```

Comment:
the data byte array is not copied here.

So , if we have a `PrestoSparkMaterializedRow r1` and convert to `PrestoSparkMutableRow r2`, and then do changes over `r2`, would that affect data in `r1`?

Reply:
As I mentioned you can think of `PrestoSparkMaterializedRow` just as a transitional think. It is removed in the last commit. You don't have to spend much time reviewing this.